File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/local/StringListConverter.java
package com.dani_chuks.andeladeveloper.starwars.data.db.local;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class StringListConverter {
    @TypeConverter
    public static List<String> stringToList(String value) {
        if (value == null) {
            return Collections.emptyList();
        }
        Type listType = new TypeToken<List<String>>() {}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromArrayList(List<String> list) {
        Gson gson = new Gson();
        return gson.toJson(list);
    }
}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/StringListConverter.kt
package com.dani_chuks.andeladeveloper.starwars.data.db

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class StringListConverter {
    @TypeConverter
    fun stringToList(value: String?): List<String> {
        if (value == null) {
            return emptyList()
        }
        val listType = object : TypeToken<List<String>>() {

        }.type
        return Gson().fromJson(value, listType)
    }

    @TypeConverter
    fun fromArrayList(list: List<String>): String {
        val gson = Gson()
        return gson.toJson(list)
    }

    interface Source<out T> {
        fun nextT(): T
    }
}
